AirCasting/SessionViews/ThresholdSettings/ThresholdsSettingsViewModel.swift

```diff
@@ -37,10 +37,13 @@ class ThresholdSettingsViewModel: ObservableObject {
             return
         }
         
-        let newValues: [Int32] = [thresholdVeryHigh, thresholdHigh, thresholdMedium, thresholdLow, thresholdVeryLow]
+        var newValues: [Int32] = [thresholdVeryHigh, thresholdHigh, thresholdMedium, thresholdLow, thresholdVeryLow]
             .map { formatter.value(from: $0) ?? 0 }
             .sorted { $0 < $1 }
         
+        // Prevents us from the situation when the user could change thresholds to have the same values
+        for value in (1...newValues.count - 1).reversed() { if newValues[value - 1] == newValues[value] { newValues[value - 1] -= 1 }}
```

Comment:
I don't like how unreadable this is. I wrote a simple extension for `Sequence` that is basically a `map` with an ability to have a peek what the next `Element` is:
```Swift
extension Sequence {
    func mapWithNext(_ map: ((Element, Element) -> Element)) -> [Element] {
        var iter = self.makeIterator()
        var prev = iter.next()
        guard prev != nil else { return [] }
        
        var pairs: [(Element, Element)] = []
        while let elem = iter.next() {
            pairs.append((prev!, elem))
            prev = elem
        }
        var transformed = pairs.map { map($0.0, $0.1) }
        transformed.append(pairs.last!.1)
        return transformed
    }
}
```

We could use it here like that:
```Swift
newValues.reversed().mapWithNext { lowerThreshold, higherThreshold in
    guard lowerThreshold != higherThreshold else { return lowerThreshold - 1.0 }
    return lowerThreshold
} 
```